models/choral.js

```diff
@@ -0,0 +1,58 @@
+var mongoose = require('./setup'),
+  ObjectId = mongoose.Schema.Types.ObjectId;
+var uniqueValidator = require('mongoose-unique-validator');
+var srs = require('secure-random-string');
+
+var childSchema = new mongoose.Schema({
+  id: { type: ObjectId, required: '{PATH} is required!' }
+});
+
+var choralSchema = new mongoose.Schema({
+  children: [childSchema],
+  userId: { type: ObjectId, required: '{PATH} is required!' },
+  choralId: { // refers to choral_id in cassandra
+    type: String,
+    required: '{PATH} is required!',
+    unique: true,
+    minLength: 128
+  },
+  func: {
+    type: String,
+    required: [
+      function() { return this.type == 'choral' },
+      '{PATH} is required for non-device chorals!'
+    ],
+    default: "function(data) { return data[0] || 1; }"
+  },
+  sampleRate: {
+    type: Number,
+    default: 5,
+    min: 1,
+  },
+  choralType: {
+    type: String,
+    enum: ['choral', 'device'],
+    required: '{PATH} is required!'
+  }
```

Comment:
Now that I see this schema, I think that it might be a good idea to be able to name the choral. This name doesnt have to be unique, but it would make it easier to see what the choral represents.